### ISSUE TITLE
torch.clamp does not clamp out of -0 from 0 when ran on the CPU

### ISSUE THREAD

**Issue by creakseek** [pytorch/pytorch#87131]
### 🐛 Describe the bug

when ran on the CPU, `torch.clamp` does not clamp out of -0 from 0 (i.e. when min is 0, it does not change -0 to 0 as it should). 
- output on cpu: -0.
- output on gpu: 0.
- output with numpy: 0.

See this code snippet to produce the bug:
```
import torch
import numpy as np

torch.manual_seed(420)
torch.cuda.manual_seed_all(420)
x = torch.tensor([-0., 0., 0.])
y = torch.clamp(x, min=0, max=1)  # prints [-0., 0., 0.]
print(y)

x = x.cuda()
y = torch.clamp(x, min=0, max=1)  # prints [0., 0., 0.]
print(y)

x = np.array([-0., 0., 0.])
y = np.clip(x, 0, 1)
print(y)  # prints [0., 0., 0.]
```

### Versions

torch 1.12.1

**Comment by gkneighb** [comment]
On torch 2.11.0 (CPU on Apple Silicon arm64) the original snippet still preserves -0: `clamp([-0., 0., 0.], min=0, max=1)` returns `[-0., 0., 0.]`.

Worth flagging that **numpy 2.4 now also returns `[-0., 0., 0.]` here** (it normalized to `[0., 0., 0.]` when this was filed in 2022) — so the numpy cross-reference cited in the OP no longer differs:

```python
import numpy as np
np.clip(np.array([-0., 0., 0.]), 0, 1)   # array([-0., 0., 0.])  on numpy 2.4
```

The cross-device CPU/CUDA divergence (if it still exists) is the remaining question; couldn't verify CUDA here.
